lib/compress.js

```diff
@@ -75,18 +75,36 @@ function Compressor(options, false_by_default) {
         screw_ie8     : false,
         drop_console  : false,
         angular       : false,
-
         warnings      : true,
-        global_defs   : {}
+        global_defs   : {},
+        passes        : 1,
     }, true);
+    this.warnings_produced = {};
 };
 
 Compressor.prototype = new TreeTransformer;
 merge(Compressor.prototype, {
     option: function(key) { return this.options[key] },
-    warn: function() {
-        if (this.options.warnings)
-            AST_Node.warn.apply(AST_Node, arguments);
+    compress: function(node) {
+        var passes = +this.options.passes || 1;
+        for (var pass = 0; pass < passes && pass < 3; ++pass) {
```

Comment:
> since pure annotations are removed on the first pass, thus removing the benefit of multiple passes

That hasn't been my experience with `#__PURE__` annotations. Test different `passes` settings in the following examples:

https://github.com/mishoo/UglifyJS2/issues/1261#issuecomment-285203500
https://github.com/mishoo/UglifyJS2/issues/1261#issuecomment-285210423
https://github.com/mishoo/UglifyJS2/issues/1261#issuecomment-285222553

Also, you can set an option to keep the pure annotations if you want uglify to emit them after minification.

Clarification: the `/*#__PURE__*/` annotation is not set on a function, but a function **call**. I had missed that in the recent README doc fix.